ecs-init/docker/docker_test.go

```diff
@@ -248,6 +251,20 @@ func validateCommonCreateContainerOptions(opts godocker.CreateContainerOptions,
 	if certDir := config.HostPKIDirPath(); certDir == "" {
 		expectedAgentBinds = expectedAgentBindsSuseUbuntuPlatform
 	}
+	// add exec binds
+	hostCapabilityExecResourcesDir := filepath.Join(hostCapabilitiesResourcesRootDir, capabilityExecName)
+	bindSourcePaths := []struct {
+		path      string
+		predicate func(os.FileInfo) bool
+	}{
+		{filepath.Join(hostCapabilityExecResourcesDir, capabilityExecHostBinRelativePath), isDir},
+		{filepath.Join(capabilityExecHostCertsDir, capabilityExecRequiredCert), isFile},
+	}
+	for _, path := range bindSourcePaths {
+		if existsAndValid, err := pathExistsAndValid(path.path, path.predicate); err == nil && existsAndValid {
```

Comment:
so, if I'm understanding correctly, for this unit test we depend on the actual files to exist in the host? 
The unit test should be mocking out external dependencies and just testing behavior. 

e.g.
* given that the cert file exists and is valid (mocked out), assert that cert-related binds were added

* given that the cert file does NOT exist, or is valid (mocked out), assert that cert-related binds were not added

Reply:
As things stand, it's hard to mock things because the real `pathExistsAndValid` is passed in `getHostConfig`. We might need to change the implementation strategy here, this is getting a bit more complicated than necessary

Reply:
good point! I am looking at these tests (that call `validateCommonCreateContainerOptions()`) as integration tests so I did not any mocks, but you are definitely right that this is by itself also hard to mock the predicate function, would love to know the alternatives